invenio_communities/notifications/builders.py

```diff
@@ -216,3 +217,73 @@ class SubCommunityDecline(SubCommunityBuilderBase):
     """Notification builder for subcommunity request decline."""
 
     type = f"{SubCommunityBuilderBase.type}.decline"
+
+
+class SubComInvitationBuilderBase(SubCommunityBuilderBase):
+    """Base notification builder for subcommunity invitation requests."""
+
+    type = "subcommunity-invitation-request"
+
+    recipients = [
+        CommunityMembersRecipient("request.created_by", roles=["owner", "manager"]),
+        CommunityMembersRecipient("request.receiver", roles=["owner", "manager"]),
+    ]
+
+
+class SubComInvitationCreate(SubComInvitationBuilderBase):
+    """Notification builder for subcommunity request creation."""
+
+    type = f"{SubComInvitationBuilderBase.type}.create"
+
+    context = [
+        EntityResolve("request"),
+        EntityResolve("request.created_by"),
+        EntityResolve("request.receiver"),
+        # EntityResolve("executing_user") creating via script only for now
+    ]
+
+    recipient_filters = [
+        UserPreferencesRecipientFilter(),
+        # TODO: We exceptionally DON'T filter out the executing user here, because we
+        # don't have a clear place where they can see the created request.
+        # See also: https://github.com/inveniosoftware/invenio-communities/issues/1158
+        # UserRecipientFilter("executing_user"),
+    ]
+
+
+class SubComInvitationAccept(SubComInvitationBuilderBase):
```

Comment:
minor: should the recipients here, only be the creator of the request, i.e. the parent community?